spectrum-offchain-cardano/src/data/cfmm_pool.rs

```diff
@@ -291,6 +291,58 @@ where
     }
 }
 
+impl MakerBalance for ConstFnPool {
+    fn balance(&self, that: Self) -> Option<(Self, Excess)> {
+        let x = self.asset_x.untag();
+        let y = self.asset_y.untag();
+        let [base, _] = order_canonical(x, y);
+        let drx = that.reserves_x.checked_sub(&self.reserves_x).map(|x| x.untag());
+        if let Some(drx) = drx {
+            // input is X
+            let trade_input = drx;
+            let side = if x == base { Side::Ask } else { Side::Bid };
+            let rebalanced = match self.swap(side.wrap(trade_input)) {
+                Next::Succ(pool) => Some(pool),
+                Next::Term(_) => None,
+            }?;
+            let excess_y = rebalanced.reserves_y.checked_sub(&that.reserves_y)?.untag();
+            let delta = if x == base {
```

Comment:
If we checked that Some(drx) == drx at 300 and verify that x is base and y is quote, why do we need another check here?